sql/analyzer/pushdown.go

```diff
@@ -700,3 +705,85 @@ func (es exprSlice) String() string {
 	}
 	return sb.String()
 }
+
+// stripDecorations removes *plan.DecoratedNode that wrap plan.ResolvedTable instances.
+// Without this step, some prepared statement reanalysis rules fail to identify
+// filter-table relationships.
+func stripDecorations(ctx *sql.Context, a *Analyzer, node sql.Node, scope *Scope) (sql.Node, error) {
+	return plan.TransformUp(node, func(node sql.Node) (sql.Node, error) {
+		switch n := node.(type) {
+		case *plan.DecoratedNode:
+			return n.Child, nil
+		default:
+			return node, nil
+		}
+	})
+
+}
+
+// unresolveTables is a quick and dirty way to make prepared statement reanalysis
+// resolve the most up-to-date table roots while preserving projections folded into
+// table scans.
+func unresolveTables(ctx *sql.Context, a *Analyzer, node sql.Node, scope *Scope) (sql.Node, error) {
+	return plan.TransformUpCtx(node, nil, func(c plan.TransformContext) (sql.Node, error) {
```

Comment:
This isn't going to hit IndexedTableAccess, and it really needs to.

Make a test for that.